README.md

```diff
@@ -304,6 +311,79 @@ total-size: false
 symlink-arrow: ⇒
 ```
 
+## Theme
+
+`lsd` can be configured with a theme file to set the colors.
+
+Theme can be configured in the [configuration file](#configuration)(color.theme),
+The valid theme configurations are:
+- `default`: the default color scheme shipped in `lsd`
+- theme-file-name(yaml): use the theme file to specify colors
+
+when configured with the `theme-file-name` which is a `yaml` file,
+`lsd` will look up the theme file in the following way:
+- relative name: check the themes under XDG Base Directory, e.g. ~/.config/lsd/themes/<theme-file-name>
+- absolute name: use the file path and name to find theme file
+
+Check [Theme file content](#theme-file-content) for details.
+
+### Theme file content
+
+Theme file use the [crossterm](https://crates.io/crates/crossterm)
+configure the colors, check [crossterm](https://docs.rs/crossterm/0.20.0/crossterm/style/enum.Color.html)
+for the supported colors.
+
+Color table: https://upload.wikimedia.org/wikipedia/commons/1/15/Xterm_256color_chart.svg
+
+Please notice that color value would ignore case, both lowercase and UPPERCASE is supported.
+
+This is the default theme scheme shipped with `lsd`.
+
+```yaml
+user: 230
+group: 187
+permissions:
+  read: Green
+  write: Yellow
+  exec: Red
+  exec-sticky: 5
+  no-access: 245
+file-type:
```

Comment:
We don't really use these colors as we are always leveraging LSCOLORS for this. We can drop this. Anything that colors files had been dropped. Right now all file base coloring is handled by LSCOLORS.

We could have kept this, but then we will miss out on being able to use LSCOLORS which has way more configuration options.

Reply:
I do really get this, do you talking about issue #549?

Reply:
We only make use of these colors when `ThemeOption::NoLscolors` is set.
This is not available to the user and is only use it tests. So there is no point in exposing these options to the user.

Reply:
do you mean the entire `file-type` section? it makes sense that we should delete it if it could not make changes, but should we add an option for configuration it later?

and you raised an important point, we can only configure the color but not the format, should we consider adding the support for font formats like `bold`, `italy`, or something else?

Reply:
If we are to let the users configure `file-type` section, then we probably will have to have a flag to disable/enable LSCOLORS, as in users ca provide a theme name as well as set lscolors to false, but that is probably an overkill for now. LSCOLORS is capable enough to handle most things.

As for `bold`, `italics`; if you are talking about for the file type ones LSCOLORS can actually handle it. `bold` and `italics` for the other sections can come in a future pr if it is really needed(which I don't really think it is).

Reply:
yes, we have the same opinion for the `file-type` section, and I prefer we would implement it later in another PR.

for the `bold`, `italics` part, It is also not that necessary for me, but what I was considering is that there would be breaking changes if we have to add it.

as now we support themes, I prefer we could use either `LSCOLORS` or `themes` for all the coloring, not mixing them.

after a quick thought, we should be able to handle the breaking changes for the `bold` part, I will delete the `file-type` first and then create 2 issues for those 2 discussions.

Reply:
The reason why I did not want to switch to using `theme` for file name coloring is becasuse LSCOLORS is way more powerful as of now. It can color based on the name and supports things like bold, italics, underline. Also, there is a lot of projects like https://github.com/trapd00r/LS_COLORS which we will miss out on if we drop that functionality. If I remember correctly, even exa seems to be doing the same.

As for the bold/italics part of the non file-type icons, I don't think we should really bother about it now. If at all that comes up, we can still support it without breaking anything in config by just checking if the value of something is just a string/number or an object which contains bold and italics keys.

Reply:
I recheck the implementation, confirm that the theme does affect the lscolors, we will always check lscolors for file-name and then theme if no lscolors are found, I will document it to readme.

https://github.com/zwpaper/lsd/blob/03f406dd7717dba2d1074d456e4a925d24c8a30b/src/color.rs#L154-L160

so that It should be necessary to leave the file-type part in the theme and then we should discuss the formatting like bold, italics, underline here https://github.com/Peltoche/lsd/issues/550, and cowork with LSCOLORS here https://github.com/Peltoche/lsd/issues/551

Reply:
As of now, we let LSCOLORS pick the default colors if we are not able to find LSCOLORS variable. This was done because we got a lot of issues saying that the colors that lsd had did not match their terminal colors. I don't think we should change that as of now.

Reply:
We can probably discuss more on how to override the use of LSCOLORS in #551 but I don't think we should be changing the behavior of picking default LSCOLORS if no env value is set.

Reply:
like I said, this theme feature does not change the origin logic picking up colors, it replaces the last default option hardcoded before.

https://github.com/Peltoche/lsd/pull/452/files?file-filters%5B%5D=.lock&file-filters%5B%5D=.md&file-filters%5B%5D=.rs#diff-12d1cf56d5d03c679a5f8f1c48cfe969a5c4276a1bee8647cc548de13f18e465L100-L111

the `LS_COLORS` env you mentioned is handled here before the theme coloring https://github.com/Peltoche/lsd/pull/452/files#diff-12d1cf56d5d03c679a5f8f1c48cfe969a5c4276a1bee8647cc548de13f18e465L87-L91

Reply:
we are not breaking anything here, the workflow is still:
1. create `lscolors` from env, and default if not found 
   ```rust
   ThemeOption::Default => Some(LsColors::from_env().unwrap_or_default())
   ```
2. check `lscolors` is here a color option for the filename(LS_COLORS env already considered in `)
3. if no lscolors are found, use the theme configurations

Reply:
@meain what do you think? this PR should be ready if we resolve this question.

Reply:
I don't think we should ship with `file-type` when that is something that the user will never be able to use. The config should not have it. Few other nitpicks as well, just to keep the flags and config consistent, `permissions` -> `permission`, `modified` -> `date`.

We should also probably also accept both `yaml` and `yml` as extensions for the theme file but that is not that big a deal.

Sorry about the nitpicks, but once this goes out, it will be really hard to change anything without breaking the config for users.

Reply:
Something along the lines of https://github.com/meain/lsd/commit/0788c6f21c9f75685f1cfbfd2e936f841f3fca47

Reply:
will do this for consistency, but `modified` seems more accurate than `date`.

Reply:
The reason why I suggested `date` instead of `modified` is so that they will be able to map it directly from the `blocks` values. All the other values are same as the one in `blocks`.